### ISSUE TITLE
Tutorial papercuts

### ISSUE THREAD

**Issue by THOUSAND-SKY** [remix-run/remix#8395]
### Reproduction

I don't have a repro, if the completed tutorial is somewhere then it wasn't linked in the tutorial. I wish it was.

I'm using vite, I was NOT using the tutorial template, actually. I did copy the `data` file.

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
    Memory: 35.75 GB / 46.95 GB
    Container: Yes
    Shell: 3.6.4 - /usr/bin/fish
  Binaries:
    Node: 18.19.0 - ~/.nix-profile/bin/node
    Yarn: 1.22.21 - ~/.local/share/pnpm/yarn
    npm: 10.2.3 - ~/.nix-profile/bin/npm
    pnpm: 8.13.1 - ~/.local/share/pnpm/pnpm
    Watchman: 4.9.0 - /usr/bin/watchman
  npmPackages:
    @remix-run/architect: ^2.4.1 => 2.4.1 
    @remix-run/css-bundle: ^2.4.1 => 2.4.1 
    @remix-run/dev: ^2.4.1 => 2.4.1 
    @remix-run/eslint-config: ^2.4.1 => 2.4.1 
    @remix-run/node: ^2.4.1 => 2.4.1 
    @remix-run/react: ^2.4.1 => 2.4.1 
    @remix-run/serve: ^2.4.1 => 2.4.1 
    vite: ^5.0.10 => 5.0.10 
```

### Used Package Manager

pnpm

### Expected Behavior

I played through the tutorial and have some quick feedback, take it as you will.

1. Edit form doesn't clear when clicking "New" while already in an edit form. I would suggest:
    ```
    <Form
      id="contact-form"
      key={contact.id} // <= Added this to reset the form on page change, dunno if there's a better way.
      method="post"
    ```
    I'd suggest/mention something, because it's quite relevant to the navigation?
2. Destroy route should have `_`: `app/routes/_sidebar.contacts.$contactId_.destroy.tsx`? [The tutorial doesn't,](https://remix.run/docs/en/main/start/tutorial#deleting-records) makes `Delete` take me to 404 page.
3. Edit page cancel navigates out of the page if landing straight on edit page (e.g. navigate to http://localhost:5173/contacts/paul-bratslavsky/edit on new tab), I would suggest the following, although I wish `navigate(-1)` didn't have that behavior out-of-the-box:
    ```
    navigate("..", {
      relative: "path",
    })
    ```
   I think this is a common mistake, and the tutorial should mention it.

### Actual Behavior

Just thought I'd do a friendly report, not expecting anything really.

Thanks for the tutorial!

PS. again, I was NOT using the tutorial template, and I went quite far off-track and my solution is quite different. However, I couldn't see a completed version linked there so I can't compare.

**Comment by brookslybrand** [comment]
Thanks so much for sharing all this feedback! I agree it would probably be good to add a full tutorial

1. Yep, agreed, this is kind of an awkward experience, I'll add a fix for this
2. Your route looks different from the tutorials (there's no `_sidebar`), so I'm wondering if that's affecting it. I went through the tutorial and deleting leading to a 404 was not a problem for me
3. I did not experience this issue, so I'm not sure what caused it. The `navigate(-1)` should be inside of an `onClick` callback, so it shouldn't be firing automatically

**Comment by THOUSAND-SKY** [comment]
> The navigate(-1) should be inside of an onClick callback, so it shouldn't be firing automatically

Yeah it's not firing automatically but if you land on that page and then click, it navigates you back to some page outside the app.

**Comment by ivan-kleshnin** [comment]
Very nice tutorial, thank you 👍 My 5 cents to improve it.

The first place where I feel disconnected is this chapter: https://remix.run/docs/en/main/start/tutorial#forms-without-navigation

It says:

>So far all of our forms have changed the URL. While these user flows are common, it's equally common to want to submit a form without causing a navigation.

> For these cases, we have [useFetcher](https://remix.run/docs/en/main/hooks/use-fetcher). It allows us to communicate with actions and loaders without causing a navigation.

Wait, haven't we explicitly navigate to new pages with `redirect`?! If I remove `redirect`, for example, on `EDIT` layout I stay on the same page. No URL is changed... So what exactly do you mean by "without causing a navigation"?!

The difference betwee `Form` and `fetcher.Form` that I personally observe, is that `navigation.state` is updated from `idle` to `loading` and back with the first and stays idle with the second. So, I assume, `Form` should work with disabled JS and `fetcher.Form` should not. But it's not necessary related to URL changes. Would be great to somehow clarify the text.

**Comment by kupuma-ru21** [comment]
Same
I'm not sure about a difference between `<Form>` and `fetcher.Form`

> [There is one key difference though, it's not a navigation, so the URL doesn't change and the history stack is unaffected.](https://remix.run/docs/en/main/start/tutorial#forms-without-navigation)

In other words, `<Form>` changes something of URL and history stack
but I couldn't find any change in [my environment](https://github.com/kupuma-ru21/my-remix-app/blob/a6d82c0cf190857a6d06341f78c04322e2057498/app/routes/contacts.%24contactId.tsx#L84)
What do the docs mean?

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2. If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.

We plan to continue to address 2 types of issues in Remix v2:

- Bugs that pose security concerns
- Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

- If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
- If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
